File: MacPopDemo/MacPopDemo/ViewController.m
//
//  ViewController.m
//  MacPopDemo
//
//  Created by alexiuce  on 2017/9/13.
//  Copyright © 2017年 alexiuce . All rights reserved.
//

#import "ViewController.h"

#import <POP.h>

@interface ViewController ()
@property (weak) IBOutlet NSBox *boxView;

@end


@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    // Do any additional setup after loading the view.
}


- (void)setRepresentedObject:(id)representedObject {
    [super setRepresentedObject:representedObject];

    // Update the view, if already loaded.
}
- (IBAction)startAnimation:(NSButton *)sender {
    
    POPBasicAnimation *baseAnimation = [POPBasicAnimation animationWithPropertyNamed:kPOPLayerPositionX];
    baseAnimation.duration = 3.0f;
    baseAnimation.toValue = @(_boxView.frame.origin.x + 200);
    
    [_boxView.layer pop_addAnimation:baseAnimation forKey:@"demox"];
    
}


@end

File: XCRefresh/XCRefresh/ViewController.swift
//
//  ViewController.swift
//  XCRefresh
//
//  Created by alexiuce  on 2017/9/20.
//  Copyright © 2017年 alexiuce . All rights reserved.
//

import Cocoa

class ViewController: NSViewController {

    @IBOutlet weak var textScrollView: NSScrollView!
    override func viewDidLoad() {
        super.viewDidLoad()

        textScrollView.xc_HeaderRefresh(nil, action: nil)
        // Do any additional setup after loading the view.
    }

    override var representedObject: Any? {
        didSet {
        // Update the view, if already loaded.
        }
    }


}

